### ISSUE TITLE
Probably improper error message when dealing with spread operator

### ISSUE THREAD

**Issue by barcicki** [Microsoft/TypeScript#5189]
Consider following example:

```js
console.log(...['Hello', 'world']);
```

It's a completely valid `console.log` usage scenario however TypeScript (@1.6.2) raises **TS2346** error: *Supplied parameters do not match any signature of call target*.

The log's signature is:
```js
declare function log(message?: any, ...optionalParams: any[]);
```

Therefore following code:

```js
console.log('Hello', ...['world']);
```

is considered valid, without any errors.

In my opinion the top example should not trigger any errors.

**Comment by DanielRosenwasser** [comment]
Thanks for filing the issue - this looks like a duplicate of #4755. The issue to appropriately address this is tracked by #4759.
